app/assets/javascripts/controllers/middleware_deployment/middleware_deployment_controller.js

```diff
@@ -0,0 +1,38 @@
+ManageIQ.angular.app.controller('middlewareAddDeploymentController', MiddlewareDeploymentCtrl);
+
+MiddlewareDeploymentCtrl.$inject = ['$scope', '$http', 'miqService'];
+
+function MiddlewareDeploymentCtrl($scope, $http, miqService) {
+  ManageIQ.angular.scope = $scope;
+
+  $scope.showListener = function () {
+    $scope.showDeployModal = true;
+  };
+
+  $scope.addDeployment = function () {
+    miqService.sparkleOn();
+    var url = '/middleware_server/add_deployment';
+    $scope.uploadFile($scope.filePath, url);
+  };
+    
+  $scope.uploadFile = function (file, uploadUrl) {
+    var fd = new FormData();
+    fd.append('file', file);
+    fd.append('id', angular.element('#server_id').val());
+    $http.post(uploadUrl, fd, {
+      transformRequest: angular.identity,
+      headers: {'Content-Type': undefined}
+    })
+    .success(function() {
+      miqService.miqFlash('success', 'Deployment initiated for selected server.');
```

Comment:
So, would it be OK to have, for a wildfly-helloworld.war deployment, for success:
**Deployment "wildfly-helloworld.war" has been initiated on this server.**
And in case of an error:
**Unable to deploy "wildfly-helloworld.war" on this server**